### PR TITLE
Made the weather report relative to the currently displayed activity's location 

### PR DESCRIPTION
**Pull request by ecornamu**
- Modified the `WeatherActivityViewModel` so that you can switch between weather for the current user location and weather for target location in latitude/longitude passed as arguments. The app therefore displays the current location weather report on the `Discover` tab, and the weather on the `More Info` screen is from the displayed activity location.
(FYI, `Montain View` is the name of the city where my emulator seems to be located, as silly as it sounds)

image

image

- The `WeatherActivityViewModel` is now also a shared view model so that the current weather isn't related to the `Composable` that builds it

- Tried to add test suites to `WeatherActivityViewModel`, but the need of an `Application` that validates that the user authorized location usage seems to make it difficult and I didn't achieve it for now. Maybe this is why the already present test is commented ? To compensate for this lack of coverage, I added tests for `WeatherRepositoryImpl` and additional coverage to `MoreInfoScreen`

### CODE REVIEW

File: app/src/test/java/com/lastaoutdoor/lasta/data/api/weather/WeatherRepositoryImplTest.kt
Comment (Thimphou) [inline app/src/test/java/com/lastaoutdoor/lasta/data/api/weather/WeatherRepositoryImplTest.kt:1]: These tests look good, maybe you could try mockking the application to achieve more test coverage?

**Review by Thimphou — CHANGES_REQUESTED** [review]
Hey Emile! 

That is some nice work overall, I'm just requesting changes for the typo which is fixable in 10 seconds, otherwise this is a great feature for our users :)

File: app/src/test/java/com/lastaoutdoor/lasta/data/api/weather/WeatherRepositoryImplTest.kt
Comment (ecornamu) [reply app/src/test/java/com/lastaoutdoor/lasta/data/api/weather/WeatherRepositoryImplTest.kt:1]: The problem is deeper than that unfortunately, the `WeatherViewModel` verifies that the user location usage is allowed. This behaviour is weird to simulate with `mockk` and I ultimately didn't achieve it.

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/moreinfo/MoreInfoScreen.kt [hunk 1]
```diff
@@ -63,16 +63,22 @@ fun MoreInfoScreen(
     goToMarker: (Activity) -> Marker,
     weather: WeatherResponse?,
     markerList: List<Marker>,
-    selectedItenary: MapItinerary?,
+    selectedItinary: MapItinerary?,
```
[the remark this answers is not in the document]
Comment (ecornamu) [reply app/src/main/java/com/lastaoutdoor/lasta/ui/screen/moreinfo/MoreInfoScreen.kt]: My bad, fixed now :)

**Review by Thimphou — APPROVED** [review]
Thanks :)

**Pull request merged**